ODIN_II/SRC/read_xml_config_file.cpp

```diff
@@ -269,5 +272,38 @@ void read_optimizations(pugi::xml_node a_node, config_t* config, const pugiutil:
             config->split_hard_adder = 1;
     }
 
+    child = get_single_child(a_node, "mix_soft_hard_blocks", loc_data, OPTIONAL);
+    if (child != NULL) {
+        // This is a temporary block that should be removed in future
+        // Mixing soft and hard blocks is likely to interfere with
+        // other optimizations, so the can_try_to_optimize vent
+        // is introduced to avoid interactions that were not thought through
+        // or were not tested.
+        config->mults_mixing_ratio = -1.0f;
+        config->mults_mixing_exact_number_of_multipliers = -1;
+        config->mix_soft_and_hard_logic = 0;
+        bool can_try_to_optimize = true;
+
+        // config_t.h file. When introducing new hard block type
+        // to perform mixed hard/soft logic design, please modify
+        // the description in config_t.h file, as well as the enum
+        // in the odin_types.h
+        prop = get_attribute(child, "multipliers", loc_data, OPTIONAL).as_string(NULL);
+        if (prop != NULL && true == can_try_to_optimize) {
+            int bit_value = (1 << mix_hard_blocks::MULTIPLIERS);
+            config->mix_soft_and_hard_logic = config->mix_soft_and_hard_logic | bit_value;
```

Comment:
can you just set the class variables here rather than the config->stuff ? it would keep things tidy, just maybe move the logic into the class such that, part of implementing the class is: write the bit that can parse its variable from the XML